quantum/pointing_device/pointing_device_drivers.c

```diff
@@ -172,15 +170,34 @@ report_mouse_t cirque_pinnacle_get_report(report_mouse_t mouse_report) {
                 }
             }
         }
-#    endif
+#        endif
     }
 
-#    ifdef POINTING_DEVICE_GESTURES_CURSOR_GLIDE_ENABLE
+#        ifdef POINTING_DEVICE_GESTURES_CURSOR_GLIDE_ENABLE
 mouse_report_update:
-#    endif
+#        endif
     mouse_report.x = report_x;
     mouse_report.y = report_y;
+#    else
+    // Scale coordinates to arbitrary X, Y resolution
+    cirque_pinnacle_scale_data(&touchData, cirque_pinnacle_get_scale(), cirque_pinnacle_get_scale());
 
+    if (touchData.valid) {
+        if (touchData.buttons & 0x01) {
+            mouse_report.buttons = pointing_device_handle_buttons(mouse_report.buttons, true, POINTING_DEVICE_BUTTON1);
+        } else {
+            mouse_report.buttons = pointing_device_handle_buttons(mouse_report.buttons, false, POINTING_DEVICE_BUTTON1);
+        }
+        if (touchData.buttons & 0x02) {
+            mouse_report.buttons = pointing_device_handle_buttons(mouse_report.buttons, true, POINTING_DEVICE_BUTTON2);
+        } else {
+            mouse_report.buttons = pointing_device_handle_buttons(mouse_report.buttons, false, POINTING_DEVICE_BUTTON2);
+        }
```

Comment:
```suggestion
mouse_report.buttons = touchData.buttons;
```
would suffice here, no?

Reply:
Maybe, but it's not specified in the datasheet as compatible with mouse report